instance/api/instance.py

```diff
@@ -47,7 +47,9 @@ def provision(self, request, pk=None):
         Start the (re-)provisioning of an instance
         """
         instance = self.get_object()
-        if instance.status not in (instance.EMPTY, instance.READY):
+        if instance.status not in (instance.EMPTY,
+                                   instance.READY,
+                                   instance.ERROR):
```

Comment:
@chromano Ah, you're right about the NEW state - that's actually indicative of a missing value for the progress field - currently we have SUCCESS and FAILED for when it has finished processing, RUNNING for when it's still in progress, but nothing for when the current state hasn't been started yet. Maybe add a NEW value to the progress field, too? 

This way we would be able to just check for what we really want to check for here, ie that no process is already RUNNING, instead of maintaining a list of states (plus, since we remove the ERROR state from this list, we'd be back to not being able to reprovision instances in error otherwise).